# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/transportbridge/MediaPipeline.h
# Commit: ceed49af0fad
# Full Hash: ceed49af0fad6ee87f64a89adb84c791a74ced14
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 2
# Description:
#   Bug 1654112 - Don't register/unregister conduit with the call wrapper in ctor/dtor. r=ng,bwc
#   
#   Doing this outside of the dtor, especially, removes the requirement of the
#   conduit having to be destroyed on the worker thread (main).
#   
# ==============================================================================

diff -r a1c16b86d3a7 -r ceed49af0fad dom/media/webrtc/transportbridge/MediaPipeline.h
--- a/dom/media/webrtc/transportbridge/MediaPipeline.h	Mon Jan 18 16:41:26 2021 +0100
+++ b/dom/media/webrtc/transportbridge/MediaPipeline.h	Mon Jan 18 17:32:49 2021 +0100
@@ -155,8 +155,6 @@
   // Gets the current time as a DOMHighResTimeStamp
   DOMHighResTimeStamp GetNow() const;
 
-  MediaSessionConduit* Conduit() const { return mConduit; }
-
   // Thread counting
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(MediaPipeline)
 
@@ -217,12 +215,14 @@
   // pipelines do not enter data into the graph under a content principal.
   virtual void MakePrincipalPrivate_s() {}
 
+ public:
+  const RefPtr<MediaSessionConduit> mConduit;
   const DirectionType mDirection;
+
+ protected:
   Atomic<size_t> mLevel;
   std::string mTransportId;
   const RefPtr<MediaTransportHandler> mTransportHandler;
-  RefPtr<MediaSessionConduit> mConduit;  // Our conduit. Written on the main
-                                         // thread. Read on STS thread.
 
   TransportLayer::State mRtpState = TransportLayer::TS_NONE;
   TransportLayer::State mRtcpState = TransportLayer::TS_NONE;
